bitcoin/core/key.py

```diff
@@ -67,7 +192,10 @@ class CECKey:
     POINT_CONVERSION_UNCOMPRESSED = 4
 
     def __init__(self):
-        self.k = _ssl.EC_KEY_new_by_curve_name(_NID_secp256k1)
+        # WTF: not sure why the cast is necessary here, but we get back an
+        # int without it, even though we set
+        # _ssl.EC_KEY_new_by_curve_name.restype above
+        self.k = ctypes.c_void_p(_ssl.EC_KEY_new_by_curve_name(_NID_secp256k1))
```

Comment:
Ah, okay, that makes perfect sense then. From <https://docs.python.org/2/library/ctypes.html#ctypes.byref>:

> Returns a light-weight pointer to obj, which must be an instance of a ctypes type.

Meaning that in Python, `ctypes.byref(thing)` is (roughly) equivalent to `&thing` in C. It's used here in places like:

```py
... _ssl.d2i_ECPrivateKey(ctypes.byref(self.k), ...)
```

Because `self.k` represents a `EC_KEY*` and the C [signature for d2i_ECPrivateKey](https://www.openssl.org/docs/manmaster/crypto/d2i_ECPrivate_key.html) is:

```c
... d2i_ECPrivateKey(EC_KEY **key, ...)
```

Hopefully that all makes sense.

In other words, now that you pointed out (and I bothered to read) the `ctypes` docs, the "cast" looks entirely appropriate. It was my mistake to think that it wasn't. I'll clean up the code (and my comment).